### PR TITLE
docs: refactor book

### LINKED ISSUE #489: Refactor mdbook
- Our book needs to be refactored
- It has been noted that our book is more so targeted towards "Developers" and not "Users"
- To fix this, we should refactor the book to streamline the content, remove unnecessary dev docs, and generally add clarity
- This is a follow up to https://github.com/FuelLabs/fuel-indexer/issues/483
- My recommendations are as follows:

- [ ] Quickstart
  - Refactored in https://github.com/FuelLabs/fuel-indexer/pull/486
- [ ] Getting Started
  - Keep the architecture diagram and update the architecture diagram
    - Show bi-directional communication (maybe use Lucid Charts?)
    - Diagram is 85% there
  - Change "System dependencies" and "application dependencies" to just "Dependencies"
    - Should include fuelup, postgres, docker, sqlite, and wasm-snip
  - [ ] Installing Rust
    - Remove this section
  - [ ] System dependencies
    - Remove this section
  - [ ] Application dependencies
    - Remove this section
      - [ ] Database
        - Remove this section
  - [ ] Add new section `Dependencies`
    - Should include Docker, Postgres, SQLite
    - Should only include `wasm-snip`
  - [ ] Configuration
    - Explain that you can use CLI args/opts OR the config file
    - Make sure the CLI args/opts are up-to-date 
    - Move this from under "Getting started"
  - [ ] A Fuel Indexer Project
    - Only explain the project insofar as the indexer is concerned (not other components, Sway, etc)
    - Move `Manifest`, `Schema`, `Module` from "Components -> Assets" to under "Fuel Indexer Project"
      - I applaud our layout but maybe making things a bit more concise/straight-to-the-point will help
- [ ] Examples
  - Keep these as is
- [ ] Plugins
  - [ ] forc-index
    - Update this with latest commands
    - Make sure to communicate that **the plugin is the primary interface to fuel indexer development**
- [ ] Components
  - [ ] Assets
    - Remove this section (and subsections)
    - [ ] Manifest
      - Remove this (moved to "Fuel Indexer Project")
    - [ ] Schema
      - Remove this (moved to "Fuel Indexer Project")
    - [ ] Module
      - Remove this (moved to "Fuel Indexer Project")
  - [ ] Database
    - Keep this and its subsections as is
  - [ ] GraphQL
    - Keep this and subsections as is
- [ ] What Can I Index?
  - Keep this and subsections as is

@deekerno @hrryndrsn 
- Feel free to chime in here. I will update this 👆 with anything you guys feel I missed or left out. 
- From here, we could put up a PR that we all co-author, and just tick each of these ☝️ boxes as we go.

### PR DESCRIPTION
**Pull request by deekerno**
Closes #489.

## Changelog
- Break up structure into _Getting Started_ and _Reference Guide_ main sections (inspiration taken from [mdBook](https://rust-lang.github.io/mdBook/format/summary.html), the program used to generate our own book)
- Add section for `forc index` and its individual subcommands
- Add a _For Contributors_ section
- Revise "A Fuel Indexer Project" to reference individual index project pieces (e.g. manifest, schema, module)

## Notes
This does not add much new information, if at all. The main purpose of this refactor was to move to a structure that is easier for us to maintain and external users to find information within.

Regarding native execution and optional schema fields, I figured that we still need to determine the tooling and messaging around each of those concepts before we consider adding them in the book.

### CODE REVIEW

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
-  - [GraphQL](./components/graphql/index.md)
-    - [Schema](./components/assets/schema.md)
-    - [API Server](./components/graphql/api-server.md)
-    - [Types](./components/database/types.md)
-    - [Directives](./components/graphql/directives.md)
-  - [Indices](./components/indices/index.md)
-    - [Registering Indices](./components/indices/registration.md)
+[Quickstart](./quickstart/index.md)
+
+# User Guide
```
Comment (ra0x3) [inline docs/src/SUMMARY.md:8]: I like the "User Guide" and "Reference Guide" breakdown (feel free to add the inspiration for that to the PR description - for others who might be curious where this breakdown comes from)

**Review by ra0x3 — CHANGES_REQUESTED** [review]
- I was gonna leave a ton of comments but I feel like that might not be the best way to review this.
- We have the new README [in this PR](https://github.com/FuelLabs/fuel-indexer/pull/492/files) and we have the new book in your/this PR.
- It's a bit of a chicken/egg thing (kinda) but we need to make sure there is parity between the book and the README
  - Obviously the book has to communicate more info, but they should be somewhat similar in their structure
-  I think the "User Guide" section looks good and simple
  - I would cleanup the "Installation" section
    - I imagine the "Installation" section would include the "Dependencies" section from the README 
- I'm "Requesting Changes" but only with regard to _how_ we should review this
  - Synchronously? That might lowkey be faster. I imagine there's gonna be  a _ton_ of feedback on this

There's obviously more to say here. But mainly want to figure out what you think might be the best method for communicating all of this feedback - keeping in mind the feedback for the README could also be related to the feedback for the book?

Overall, good stuff (docs are hard 🥲 ) the book already looks _much_ better and more streamlined.

**Comment by deekerno** [comment]
> * I was gonna leave a ton of comments but I feel like that might not be the best way to review this.
> 
>     * We have the new README [in this PR](https://github.com/FuelLabs/fuel-indexer/pull/492/files) and we have the new book in your/this PR.
> 
>     * It's a bit of a chicken/egg thing (kinda) but we need to make sure there is parity between the book and the README
>       
>       * Obviously the book has to communicate more info, but they should be somewhat similar in their structure
> 
>     * I think the "User Guide" section looks good and simple
> 
>     * I would cleanup the "Installation" section
>       
>       * I imagine the "Installation" section would include the "Dependencies" section from the README
> 
>     * I'm "Requesting Changes" but only with regard to _how_ we should review this
>       
>       * Synchronously? That might lowkey be faster. I imagine there's gonna be  a _ton_ of feedback on this
> 
> 
> There's obviously more to say here. But mainly want to figure out what you think might be the best method for communicating all of this feedback - keeping in mind the feedback for the README could also be related to the feedback for the book?
> 
> Overall, good stuff (docs are hard 🥲 ) the book already looks _much_ better and more streamlined.

I don't mind reviewing the README first to determine what should go in and then making sure that the book achieves parity afterwards. We'll be able to lift a lot of stuff from the README and include it in the book without much modification.

As far as reviewing it, I think that synchronous communication would be faster and I don't want to be stuck on docs much longer. 😅 I'll hit you up out-of-band to set that up.

**Review by ra0x3 — APPROVED** [review]
A true labor of love. I think this lays the foundation for a great book going forward 🙏

**Pull request merged**